servicetalk-client-api-internal/src/main/java/io/servicetalk/client/api/internal/AbstractRequestConcurrencyController.java

```diff
@@ -45,7 +45,7 @@
         // Subscribe to onClosing() before maxConcurrency, this order increases the chances of capturing the STATE_QUIT
         // before observing 0 from maxConcurrency which could lead to more ambiguous max concurrency error messages for
         // the users on connection tear-down.
-        toSource(onClosing.publishAndSubscribeOnOverride(immediate())).subscribe(new CompletableSource.Subscriber() {
+        toSource(onClosing.publishAndSubscribeOn(immediate())).subscribe(new CompletableSource.Subscriber() {
```

Comment:
Same question, will this `immediate()` change anything if the callback is already offloaded?

Because `onClosing()` is an internal API, I think we can relax it and remove offloading. Not expected that users will use it. If they will cast the types to access this method, they have to take care of offloading. WDYT?

Reply:
As in the other case it can be removed and it is especially irrelevant if `onClosing` is not offloaded.

Reply:
I also removed it in `AbstractReservableRequestConcurrencyController`

Reply:
WDYT about removing the `onCloseNoOffload()` and just documenting that `onClose()` needs to be offloaded if shared with application layer?

Reply:
`onClose()` is quite popular in users code. They frequently intercept connection closure or client/server shutdown. I would say it's risky to disable offloading for this callback.